### PR TITLE
Fix for windows(mingw)

### PR DESCRIPTION
**Pull request by takahashim**
This PR makes build success on Windows.

* `File.expand_path`: support drive letter and `ALT_SEPARATOR`
* `File.dirname`: support `ALT_SEPARATOR`
* `File.basename`: ditto.
* `IO.popen`: raise `NotImplementedError`
* `IO.pipe`: ditto.
* `` `cmd` ``:  ditto.
* `File#flock`: ditto.
* `FileTest.pipe?`: ditto.
* `FileTest.symlink?`: ditto.
* `FileTest.socket?`: ditto.

### CODE REVIEW

File: test/file.rb [hunk 1]
```diff
@@ -76,8 +80,13 @@
 end
 
 assert('File.realpath') do
-  usrbin = IO.popen("cd bin; /bin/pwd -P") { |f| f.read.chomp }
-  assert_equal usrbin, File.realpath("bin")
+  if File.const_defined?(:ALT_SEPARATOR) && File::ALT_SEPARATOR
```
Comment (tsahara) [inline test/file.rb]: Can it be just `if File::ALT_SEPARATOR`?

Comment (takahashim) [inline test/file.rb]: You're right.  I have used `const_defined?` before #55 was merged.  But now we can remove it.

Comment (tsahara) [inline test/file.rb]: Ok, I have removed it (https://github.com/iij/mruby-io/commit/99e10b8dc44cd7e362f23706cc94a3e0ac840ed7).  Thanks.

Comment (takahashim) [inline test/file.rb]: :+1: Thanks, too!

**Pull request merged**